graphql_compiler/schema_transformation/utils.py

```diff
@@ -119,15 +135,49 @@ class CascadingSuppressionError(SchemaTransformError):
     """
 
 
-_alphanumeric_and_underscore = frozenset(six.text_type(string.ascii_letters + string.digits + "_"))
+_alphanumeric_and_underscore: FrozenSet[str] = frozenset(
+    six.text_type(string.ascii_letters + string.digits + "_")
+)
 
 
 # String representations for the GraphQL built-in scalar types
 # pylint produces a false positive-- see issue here: https://github.com/PyCQA/pylint/issues/3743
-builtin_scalar_type_names = frozenset(specified_scalar_types.keys())  # pylint: disable=no-member
-
-
-def check_schema_identifier_is_valid(identifier):
+builtin_scalar_type_names: FrozenSet[str] = frozenset(
+    specified_scalar_types.keys()  # pylint: disable=no-member
+)
+
+
+# Union of classes of nodes to be renamed or suppressed by an instance of RenameSchemaTypesVisitor.
+# Note that RenameSchemaTypesVisitor also has a class attribute rename_types which parallels the
+# classes here. This duplication is necessary due to language and linter constraints-- see the
+# comment in the RenameSchemaTypesVisitor class for more information.
+# Unfortunately, RenameTypes itself has to be a module attribute instead of a class attribute
+# because a bug in flake8 produces a linting error if RenameTypes is a class attribute and we type
+# hint the return value of the RenameSchemaTypesVisitor's
+# _rename_or_suppress_or_ignore_name_and_add_to_record() method as RenameTypes. More on this here:
+# https://github.com/PyCQA/pyflakes/issues/441
+RenameTypes = Union[
+    EnumTypeDefinitionNode,
+    InterfaceTypeDefinitionNode,
+    NamedTypeNode,
+    ObjectTypeDefinitionNode,
+    UnionTypeDefinitionNode,
+]
+RenameTypesT = TypeVar("RenameTypesT", bound=RenameTypes)
+
+# For the same reason as with RenameTypes, these types have to be written out explicitly instead of
+# relying on allowed_types in get_copy_of_node_with_new_name.
+# Unlike RenameTypes, RenameNodes also includes fields because it's used in the function
+# get_copy_of_node_with_new_name which rename_query depends on to rename the root field in a query.
+# Meanwhile, RenameTypes applies only for rename_schema and field renaming in the schema is not
+# implemented yet.
+RenameNodes = Union[
+    RenameTypes, FieldNode, FieldDefinitionNode,
+]
+RenameNodesT = TypeVar("RenameNodesT", bound="RenameNodes")
```

Comment:
Any reason to use `"RenameNodes"` here as a forward reference even though `RenameNodes` is defined on the previous line?

Reply:
See below

Reply:
Ah, gotcha-- will fix